packages/atlas-service/src/store/atlas-signin-reducer.spec.ts

```diff
@@ -337,6 +344,9 @@ describe('atlasSignInReducer', function () {
         expect(err).to.have.property('message', 'Aborted from outside');
       }
       expect(store.getState()).to.have.property('state', 'canceled');
+
+      // Ensure that we are not leaving a dangling store operation that would conflict with our mocks being reset.
+      await signInCalled;
```

Comment:
@gribnoysup Just fyi in case you run across this later … I pushed this because it seems that mocha [internally also uses](https://github.com/mochajs/mocha/blob/37deed262d4bc0788d32c66636495d10038ad398/browser-entry.js#L116) `setTimeout`, which is really the only relative timing case where `setTimeout` and `setImmediate` differ (i.e. they have their own lists for callbacks rather than sharing one).

Without this fix, what happened was that the test finished before [the core logic](https://github.com/mongodb-js/compass/blob/b57b2505f156d942f97d32927d3c8676c14e9489/packages/atlas-service/src/store/atlas-signin-reducer.ts#L473) in `signIn()` ran, so the `sandbox.reset()` in the `afterEach()` hook ran _before_ the action called `atlasService.signIn()`, which then returned `undefined` (because the sandbox was reset) and then crashed because `undefined` is not an expected return value for `atlasService.signIn()`.

It would have been possible to work around this by changing the timing again slightly, but I think it’s okay to go this way here and make the test a bit more robust with regard to these subtle timing differences.